File: DiscoDruid/app/src/main/java/com/andruid/magic/discodruid/viewholder/ArtistViewHolder.java
package com.andruid.magic.discodruid.viewholder;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.andruid.magic.discodruid.R;
import com.andruid.magic.discodruid.model.Artist;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ArtistViewHolder extends RecyclerView.ViewHolder {
    @BindView(R.id.artist_name) TextView artistTV;
    @BindView(R.id.artist_song_count) TextView songCountTV;
    @BindView(R.id.artist_album_count) TextView albumCountTV;

    public ArtistViewHolder(@NonNull View itemView) {
        super(itemView);
        ButterKnife.bind(this,itemView);
    }

    public void setItem(Artist artist){
        artistTV.setText(artist.getArtist());
        songCountTV.setText(String.valueOf(artist.getTracksCount()));
        albumCountTV.setText(String.valueOf(artist.getAlbumsCount()));
    }
}

File: DiscoDruid/app/src/main/java/com/andruid/magic/discodruid/ui/viewholder/ArtistViewHolder.kt
package com.andruid.magic.discodruid.ui.viewholder

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.andruid.magic.discodruid.R
import com.andruid.magic.discodruid.databinding.LayoutArtistBinding
import com.andruid.magic.discodruid.util.getAlbumArtBitmap
import com.andruid.magic.medialoader.model.Artist
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class ArtistViewHolder(private val binding: LayoutArtistBinding) :
    RecyclerView.ViewHolder(binding.root) {
    companion object {
        fun from(parent: ViewGroup): ArtistViewHolder {
            val inflater = LayoutInflater.from(parent.context)
            val binding = DataBindingUtil.inflate<LayoutArtistBinding>(
                inflater,
                R.layout.layout_artist,
                parent,
                false
            )
            return ArtistViewHolder(binding)
        }
    }

    fun bind(context: Context, scope: CoroutineScope, artist: Artist) {
        binding.artist = artist

        scope.launch {
            val bitmap = withContext(Dispatchers.IO) { context.getAlbumArtBitmap(artist.albumId) }
            binding.thumbnailIV.setImageBitmap(bitmap)
        }

        binding.executePendingBindings()
    }
}